src/gui/components/TextInput.zig

```diff
@@ -514,7 +520,17 @@ pub fn render(self: *TextInput, mousePosition: Vec2f) void {
 			draw.setColor(0x440000ff);
 			self.textBuffer.drawSelection(textPos, @min(selectionStart, cursor), @max(selectionStart, cursor));
 		}
-		draw.setColor(0xff000000);
-		draw.line(cursorPos, cursorPos + Vec2f{0, 16});
+
+		const milliTime = std.time.milliTimestamp();
+		if(self.lastBlinkTime + blinkDurationMs <= milliTime) {
```

Comment:
We don't want bugs/crashes on integer overflow, so please use arithmetic that is aware of it:
```suggestion
		if(milliTime -% self.lastBlinkTime > blinkDurationMs) {
```